install/kubernetes/cilium/templates/spire/agent/daemonset.yaml

```diff
@@ -29,15 +29,21 @@ spec:
       serviceAccountName: {{ .Values.authentication.mutual.spire.install.agent.serviceAccount.name }}
       initContainers:
         - name: init
-          image: docker.io/library/busybox:1.35.0@sha256:223ae047b1065bd069aac01ae3ac8088b3ca4a527827e283b85112f29385fb1b
+          image: {{ include "cilium.image" .Values.authentication.mutual.spire.install.initImage | quote }}
+          imagePullPolicy: {{ .Values.authentication.mutual.spire.install.initImage.pullPolicy }}
           command:
             - /bin/sh
             - -c
             - |
               {{- tpl (.Files.Get "files/spire/wait-for-spire.bash") . | nindent 14 }}
       containers:
         - name: spire-agent
+          {{- if eq (typeOf .Values.authentication.mutual.spire.install.agent.image) "string" }}
```

Comment:
The PR descriptions making this an "image" and "tag" value, similar to other components?

Reply:
yes
the common way of cilium images use "tag" and "repository" things to define the image
I think it conforms to best practices, especially for offline environment, the administrator could just set helm option --xx. repository=offline.io/cilium , and is not necessary to care about the "tag" for cilium v1.4 or v1.5, and just keep the best matched and recommended tag in the chart

Therefore, the original definition of spire image does not conform what other cilium image does, and that is what this PR want to optimize

Reply:
and why it is compatible with the string type here , is owing to [comment](https://github.com/cilium/cilium/pull/27621#discussion_r1303870784)

Reply:
Sorry, some of the confusion may be because I added the release note. I will say though that these settings should conform to the same configuration format that is exposed for components like hubble-ui.